actors/builtin/reward/reward_calc.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python3
+
+# This file can be used to recalculate the reward constants in the code when
+# changing the reward factor and/or the epoch duration in seconds.
+
+from decimal import Decimal
+import decimal
+
+# Number of seconds per epoch.
+EPOCH_DURATION_SECONDS = 30
+
+# Growth factor per year. Currently 200%.
+GROWTH_FACTOR = 2
+
+# Precision factor.
+Q128 = 2**128
+
+# Set the precision to enough digits to store (Q128 * Q128).
+#
+# This gives us wolfram-alpha level precision.
+decimal.getcontext().prec=int(Decimal(Q128**2).log10().to_integral_value(decimal.ROUND_CEILING))
+
+def epochs_in_year() -> Decimal:
+    return Decimal(60*60*365*24)/Decimal(EPOCH_DURATION_SECONDS)
+
+def q128(val, round) -> str:
+    return str((Q128 * val).to_integral_value(round))
+
+# exp(ln[1 + 200%] / epochsInYear)
+def baseline_exponent() -> Decimal: 
+    return (Decimal(1 + GROWTH_FACTOR).ln() / epochs_in_year()).exp()
+
+def baseline_exponent_q128() -> str: 
+    return q128(baseline_exponent(), decimal.ROUND_HALF_UP)
+
+# ln(2) / (6 * epochsInYear)
+def reward_lambda() -> Decimal: 
+    return Decimal(GROWTH_FACTOR).ln() / (6 * epochs_in_year())
```

Comment:
This should not depend on the baseline GROWTH_FACTOR, it is just ln(2) here (as 6 years is a half-life). It’s a coincidence that GROWTH_FACTOR happens to be 2.

Reply:
Ah, got it.